test/e2e/generate_kube_test.go

```diff
@@ -776,8 +776,8 @@ var _ = Describe("Podman generate kube", func() {
 		Expect(pod.Spec.DNSConfig.Nameservers).To(ContainElement("8.8.8.8"))
 		Expect(pod.Spec.DNSConfig.Searches).To(ContainElement("foobar.com"))
 		Expect(len(pod.Spec.DNSConfig.Options)).To(BeNumerically(">", 0))
-		Expect(pod.Spec.DNSConfig.Options[0].Name).To(Equal("color"))
-		Expect(*pod.Spec.DNSConfig.Options[0].Value).To(Equal("blue"))
+		Expect(pod.Spec.DNSConfig.Options[0]).To(HaveField("Name", "color"))
+		Expect(pod.Spec.DNSConfig.Options[0]).To(HaveField("Value", "blue"))
```

Comment:
This is really annoying:
```
  Value for field 'Value' failed to satisfy matcher.
  Expected
      <*string | 0xc000504380>: blue
  to equal
      <string>: blue
```
I've tried `BeEquivalent()` and `BeEqual()` and a variety of other dead ends. Does anyone know if there's an incantation for dereferencing `Value`, or for telling `HaveField()` to just shut up and Do What I Mean?

Reply:
The below diff turned it green on my end:

```diff
diff --git a/test/e2e/generate_kube_test.go b/test/e2e/generate_kube_test.go                         
index f58b121f3070..31f2ea16e927 100644                                                              
--- a/test/e2e/generate_kube_test.go                                                                 
+++ b/test/e2e/generate_kube_test.go                                                                 
@@ -777,7 +777,8 @@ var _ = Describe("Podman generate kube", func() {                                
                Expect(pod.Spec.DNSConfig.Searches).To(ContainElement("foobar.com"))                 
                Expect(len(pod.Spec.DNSConfig.Options)).To(BeNumerically(">", 0))                    
                Expect(pod.Spec.DNSConfig.Options[0]).To(HaveField("Name", "color"))                 
-               Expect(pod.Spec.DNSConfig.Options[0]).To(HaveField("Value", "blue"))                 
+               s := "blue"                                                                          
+               Expect(pod.Spec.DNSConfig.Options[0]).To(HaveField("Value", &s))                     
        })                                                                                           
                                                                                                     
        It("podman generate kube multiple container dns servers and options are cumulative", func() {
```

Probably worth adding a helper function somewhere (`utils` package?) for such cases:
```golang
func StringToPointer(s string) *string {
     return &s
}
```